pkg/registry/apis/provisioning/repository/github.go

```diff
@@ -648,6 +653,22 @@ func (r *githubRepository) Process(ctx context.Context, logger *slog.Logger, wra
 				continue
 			}
 		}
+
+		// TODO: why having to fetch this one again?
+		resource, err := r.iface.Get(ctx, r.config.Name, metav1.GetOptions{})
```

Comment:
You shouldn't have to. Just clone the repository the GH repo contains and set its status. The `UpdateStatus` function ignores the spec, so if that's updated, that's OK

Reply:
it's not of type `unstructured.Unstructured`. How do I convert it back?

Reply:
Ahh... okay, then you do have to 😄